src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx

```diff
@@ -1370,6 +1370,15 @@
   <data name="EnumShouldNotHaveDuplicatedValuesTitle" xml:space="preserve">
     <value>Enums values should not be duplicated</value>
   </data>
+  <data name="AvoidExcessiveParametersOnGenericTypesDescription" xml:space="preserve">
+    <value>The more type parameters a generic type contains, the more difficult it is to know and remember what each type parameter represents.</value>
+  </data>
+  <data name="AvoidExcessiveParametersOnGenericTypesMessage" xml:space="preserve">
+    <value>Consider a design where '{0}' has no more than 2 type parameters.</value>
```

Comment:
Just in case we decide to make the limit configurable in future.
```suggestion
    <value>Consider a design where '{0}' has no more than {1} type parameters.</value>
```